.travis.yml

```diff
@@ -13,25 +9,61 @@ addons:
     - xmlto
     - valgrind
 
-script:
-  - # ============ Build from config ===============
-  - cp contrib/config.make .
-  - make all-debug
-  - make update-docs && git diff --exit-code
-  - make test
-  - make test TEST_OPTS=valgrind
-  - if [ $CC = clang ]; then make test-address-sanitizer; fi
-  - make DESTDIR=/tmp/bare-destdir install install-doc
-  - make DESTDIR=/tmp/bare-destdir uninstall
-  - test ! -d /tmp/bare-destdir
-  - make prefix=/tmp/bare-prefix install install-doc
-  - make prefix=/tmp/bare-prefix uninstall
-  - test ! -d /tmp/bare-prefix
-  - make distclean
-  - # ============ Build using autoconf ============
-  - make dist
-  - ./configure --prefix=/tmp/conf-prefix
-  - make V=1 TEST_SHELL=bash all test install install-doc
-  - make uninstall
-  - test ! -d /tmp/conf-prefix
-  - make clean
+matrix:
+  include:
+  - compiler: clang
+    script:
+    - # ============ Build from config ===============
+    - cp contrib/config.make .
+    - make all-debug
+    - make update-docs && git diff --exit-code
+    - make test
+    - make test TEST_OPTS=valgrind
+    - if [ $CC = clang ]; then make test-address-sanitizer; fi
+    - make DESTDIR=/tmp/bare-destdir install install-doc
+    - make DESTDIR=/tmp/bare-destdir uninstall
+    - test ! -d /tmp/bare-destdir
+    - make prefix=/tmp/bare-prefix install install-doc
+    - make prefix=/tmp/bare-prefix uninstall
+    - test ! -d /tmp/bare-prefix
+    - make distclean
+  - compiler: clang
+    script:
+    - # ============ Build using autoconf ============
+    - make dist
+    - ./configure --prefix=/tmp/conf-prefix
+    - make V=1 TEST_SHELL=bash all test install install-doc
+    - make uninstall
+    - test ! -d /tmp/conf-prefix
+    - make clean
+  - compiler: gcc
+    script:
+    - # ============ Build from config ===============
+    - cp contrib/config.make .
+    - make all-debug
+    - make update-docs && git diff --exit-code
+    - make test
+    - make test TEST_OPTS=valgrind
+    - if [ $CC = clang ]; then make test-address-sanitizer; fi
+    - make DESTDIR=/tmp/bare-destdir install install-doc
+    - make DESTDIR=/tmp/bare-destdir uninstall
+    - test ! -d /tmp/bare-destdir
+    - make prefix=/tmp/bare-prefix install install-doc
+    - make prefix=/tmp/bare-prefix uninstall
+    - test ! -d /tmp/bare-prefix
+    - make distclean
+  - compiler: gcc
+    script:
+    - # ============ Build using autoconf ============
+    - make dist
+    - ./configure --prefix=/tmp/conf-prefix
+    - make V=1 TEST_SHELL=bash all test install install-doc
+    - make uninstall
+    - test ! -d /tmp/conf-prefix
+    - make clean
+  - os: osx
+    compiler: clang
+    script: brew install --HEAD tig
```

Comment:
I don't think it makes sense to build on osx and less so using brew. In my experience, osx builds are slow, and we don't check compatibility for other package managers.

Reply:
`osx` builds are definitely slow, but i think that you have a significant user base in osx (https://bintray.com/homebrew/bottles/tig#statistics). when `brew` bottles (the binary releases) are built, no tests are run. just a way to get some sanity.

i'll remove if you think it's best (you can also make them `allow_failures`), but i think it gets more coverage for free from probably the most popular `osx` package manager

Reply:
Wow, cool. Didn’t know about those statistics. OK let’s keep one osx build then. However, then I wonder if `—head` will build the current branch. If this is not the case then the osx build should only trigger when the master branch changes. WDYT?

Reply:
your right, `--HEAD` builds `master` only. i'll throw that in and get the other changes done